backend/node/base.py

```diff
@@ -83,7 +89,28 @@ async def run_contract(
     def get_contract_data(
         self, code: str, state: str, method_name: str, method_args: list
     ):
-        return self.genvm.get_contract_data(code, state, method_name, method_args)
+        output_buffer = io.StringIO()
+
+        result = GenVM.get_contract_data(
+            code,
+            state,
+            method_name,
+            method_args,
+            self.contract_snapshot_factory,
+            output_buffer,
+        )
+
+        if self.genvm.contract_runner.mode == ExecutionMode.LEADER:
```

Comment:
@AgustinRamiroDiaz this will conflict with https://github.com/yeagerai/genlayer-simulator/pull/540